api/src/main/java/org/openmrs/module/ugandaemrreports/reports2019/SetupPNCRegister2019.java

```diff
@@ -0,0 +1,210 @@
+import org.openmrs.module.reporting.report.ReportDesign;
+import org.openmrs.module.reporting.report.definition.ReportDefinition;
+import org.openmrs.module.ugandaemrreports.data.converter.*;
+import org.openmrs.module.ugandaemrreports.definition.data.definition.CalculationDataDefinition;
+import org.openmrs.module.ugandaemrreports.library.Cohorts;
+import org.openmrs.module.ugandaemrreports.library.DataFactory;
+import org.openmrs.module.ugandaemrreports.reporting.calculation.anc.AgeLimitCalculation;
+import org.openmrs.module.ugandaemrreports.reporting.calculation.anc.PersonAddressCalculation;
+import org.openmrs.module.ugandaemrreports.reporting.calculation.anc.WhoCd4VLCalculation;
+import org.openmrs.module.ugandaemrreports.reporting.calculation.pnc.RtwRfwCalculation;
+import org.openmrs.module.ugandaemrreports.reporting.dataset.definition.SharedDataDefintion;
+import org.openmrs.module.ugandaemrreports.reporting.metadata.Dictionary;
+import org.openmrs.module.ugandaemrreports.reports.UgandaEMRDataExportManager;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Properties;
+
+/**
+ */
+@Component
+public class SetupPNCRegister2019 extends UgandaEMRDataExportManager {
+
+    @Autowired
+    private DataFactory df;
+
+    @Autowired
+    SharedDataDefintion sdd;
+
+    /**
+     * @return the uuid for the report design for exporting to Excel
+     */
+    @Override
+    public String getExcelDesignUuid() {
+        return "d6ab98f5-570d-45d6-ac57-bd143b162948";
+    }
+
+    @Override
+    public List<ReportDesign> constructReportDesigns(ReportDefinition reportDefinition) {
+        List<ReportDesign> l = new ArrayList<ReportDesign>();
+        l.add(buildReportDesign(reportDefinition));
+        return l;
+    }
+
+    /**
+     * Build the report design for the specified report, this allows a user to override the report design by adding properties and other metadata to the report design
+     *
+     * @param reportDefinition
+     * @return The report design
+     */
+    @Override
+    public ReportDesign buildReportDesign(ReportDefinition reportDefinition) {
+        ReportDesign rd = createExcelTemplateDesign(getExcelDesignUuid(), reportDefinition, "PNCRegister2019.xls");
+        Properties props = new Properties();
+        props.put("repeatingSections", "sheet:1,row:8-10,dataset:PNC");
+        props.put("sortWeight", "5000");
+        rd.setProperties(props);
+        return rd;
+    }
+
+    @Override
+    public String getUuid() {
+        return "ffed8404-7f81-4c9e-a9eb-db2673397cc0";
+    }
+
+    @Override
+    public String getName() {
+        return "Integrated Postnatal Register 2019";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Contains mothers information after delivery";
```

Comment:
@solemabrothers just use PostNatal Register the register also includes baby information

Reply:
Done